infra/macos/1-create-box/README.md

```diff
@@ -0,0 +1,173 @@
+exist.
+
+Note that if you have Homebrew installed, you also already have the CLI tools
+and can skip the next step.
+
+## XCode CLI Tools
+
+Within a graphical environment, run:
+
+```bash
+xcode-select --install
+```
+
+This will open a graphical dialog to install the XCode Command-Line Tools.
+
+## VirtualBox
+
+Download and install
+[https://download.virtualbox.org/virtualbox/6.1.4/VirtualBox-6.1.4-136177-OSX.dmg](VirtualBox),
+including the
+[https://download.virtualbox.org/virtualbox/6.1.4/Oracle_VM_VirtualBox_Extension_Pack-6.1.4.vbox-extpack](extension
+pack).
+
+Alternatively, it can be installed from Homebrew using:
+
+```bash
+brew cask install virtualbox
+```
+
+**This will require sudo access.**
+
+The extension pack has to be manually added after the installation of
+VirtualBox; this can be done through the UI, or running:
+
+```bash
+sudo VBoxManage extpack install /path/to/download/Oracle_VM_VirtualBox_Extension_Pack-6.1.4.vbox-extpack
+```
+
+These instructions have been tested against 6.1.4.
+
+## Vagrant
+
+To easily manage the guest VM from scripts, we will be using [Vagrant](https://releases.hashicorp.com/vagrant/2.2.7/vagrant_2.2.7_x86_64.dmg).
+
+Alternatively, installing with Homebrew:
+
+```
+brew cask install vagrant
+```
+
+These instructions have been tested with 2.2.7.
+
+## rbenv
+
+The underlying scripts to create the "blank" macOS guest machine are written in
+Ruby, so we need to install some version of Ruby. Ruby does come with macOS,
+and has installers in Homebrew, but for the sake of stability we are going to
+go with `rbenv`, so we can pin down the exact Ruby version we use. This also
+has the advantage of not needing root access to install the `macinbox` gem.
+(Though `sudo` still needs to be used to run it.)
+
+On the host machine, for the user that is going to manage the guest VM, add the
+following to the relevant shell init file (`~/.zshrc` by default):
+
+```bash
+if [ -d $HOME/.rbenv/shims ]; then
+    export PATH="$HOME/.rbenv/shims:$PATH"
+fi
+if [ -d $HOME/.rbenv/bin ]; then
+    export PATH="$HOME/.rbenv/bin:$PATH"
+fi
+```
+
+then run the following:
+
+```bash
+git clone https://github.com/rbenv/rbenv.git ~/.rbenv
+cd ~/.rbenv
+git checkout c6324ff45af33a194f5658a1c6322a94da145f98
+```
+
+Note: this is the commit I tested these instructions with; I have no reason to
+believe the `master` branch is bad.
+
+In a new terminal (to pick up the new PATH), run:
+
+```bash
+mkdir -p "$(rbenv root)"/plugins
+git clone https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build
+cd "$(rbenv root)/plugins/ruby-build"
+git checkout 0ef5e055659230e2fb2ae9b0928f70dc27c1c136
+```
+
+where, again, this is the commit I tested with, so I am providing it for
+maximum reproducibility, but I have no reason to mistrust the master branch.
+
+Finally, run:
+```bash
+rbenv install 2.7.0
+```
+
+which installs Ruby 2.7.0 for the current user.
```

Comment:
Needs to include a command to change default version to 2.7.0 or it still uses system version and you get a WritePermission error on next step

Reply:
This should be done the modification to your shell init script on line 97 above. Have you tried in a new terminal?

Reply:
Ah, I see. It looks like I forgot a file. Added. Thanks for pointing out!